assets/src/blocks/TopicLink/TopicLinkEditor.js

```diff
@@ -0,0 +1,255 @@
+/* eslint-disable no-shadow */
+
+const {useSelect} = wp.data;
+const {
+  BlockControls,
+  MediaUpload,
+  MediaUploadCheck,
+  InspectorControls,
+} = wp.blockEditor;
+const {
+  SelectControl,
+  PanelBody,
+  ToolbarGroup,
+  ToolbarButton,
+  FocalPointPicker,
+  Button,
+} = wp.components;
+const {__} = wp.i18n;
+
+/**
+ * TopicLinkEditor component for managing the Topic Link block in the editor.
+ *
+ * @param {Object} props - Component properties.
+ * @param {Object} props.attributes - Block attributes.
+ * @param {boolean} props.isSelected - Indicates if the block is selected.
+ * @param {Function} props.setAttributes - Function to update block attributes.
+ * @return {JSX.Element} The Topic Link Editor component.
+ */
+export const TopicLinkEditor = ({
+  attributes,
+  isSelected,
+  setAttributes,
+}) => {
+  const {
+    categoryId,
+    focal_points,
+    imageId: customImageId,
+    imageUrl: customImageFromId,
+  } = attributes;
+
+  const {
+    categoriesList,
+    imageId,
+    imageUrl,
+    imageAlt,
+    currentPostCategories,
+  } = useSelect(select => {
+    const categoriesList = [].concat(
+      select('core').getEntityRecords('taxonomy', 'category', {
+        hide_empty: true,
+        per_page: -1,
+      }) || []
+    );
+
+    const customImage = customImageId && select('core').getMedia(customImageId);
+    const customImageFromId = customImage?.source_url;
+
+    const imageId = customImageId;
+    const imageUrl = customImageFromId;
+    const imageAlt = customImage?.alt_text;
+    const currentPostCategories = select('core/editor').getCurrentPost().categories || [];
+
+    return {
+      categoriesList,
+      imageId,
+      imageUrl,
+      imageAlt,
+      currentPostCategories,
+    };
+  }, [categoryId, customImageId, customImageFromId]);
+
+  if (!categoriesList.length) {
+    return __('Populating block\'s fields…', 'planet4-blocks-backend');
+  }
+
+  /**
+   * Sets the block's category based on available categories or current post categories.
+   * @return {Object} The selected category object.
+   */
+  const setBlockCategory = () => {
+    const postCategory = categoriesList.find(category => category.id === currentPostCategories[0]);
+    const blockCategory = categoriesList.find(category => category.id === categoryId);
+
+    let selectedCategory = null;
+
+    if (blockCategory) {
+      selectedCategory = blockCategory;
+      setAttributes({categoryId: parseInt(blockCategory.id)});
+    } else if (postCategory) {
+      selectedCategory = postCategory;
+      setAttributes({categoryId: parseInt(postCategory.id)});
+    } else {
+      selectedCategory = categoriesList[0];
+      setAttributes({categoryId: parseInt(categoriesList[0].id)});
+    }
+
+    setAttributes({categoryLink: selectedCategory?.link || ''});
+    setAttributes({selectedCategory: selectedCategory?.name || ''});
```

Comment:
Maybe this attribute should be named `categoryName` instead of `selectedCategory`?

Reply:
I think it makes sense to name it `selectedCategory` to differentiate the post category from the block category. If you feel this is not clear enough let me know and I'll change it.

Reply:
Ah, I see your reasoning, personally if I see `selectedCategory` I'm thinking it's an ID. And since we have `categoryLink` for example as other attribute I think it would be more consistent to use `categoryName`...
It's not a huge deal, I'm fine with either implementation!

Reply:
Variable name was changed.